src/core/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,491 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import cx from 'classnames';
+import React from 'react';
+import { useTheme } from '../utils/hooks/useTheme';
+import '@itwin/itwinui-css/css/time-picker.css';
+import { StylingProps } from '../utils/props';
+
+const isSameHour = (
+  date1: Date,
+  date2: Date | undefined,
+  meridiem?: MeridiemType,
+) => {
+  const adjustedHours = meridiem
+    ? formatHourFrom12(date1.getHours(), meridiem)
+    : date1.getHours();
+  if (!!meridiem) {
+    return !!date2 && adjustedHours % 12 === date2.getHours() % 12;
+  }
+  return !!date2 && adjustedHours === date2.getHours();
+};
+
+const isSameMinute = (date1: Date, date2: Date | undefined) => {
+  return !!date2 && date1.getMinutes() === date2.getMinutes();
+};
+
+const isSameSecond = (date1: Date, date2: Date | undefined) => {
+  return !!date2 && date1.getSeconds() === date2.getSeconds();
+};
+
+const isSameMeridiem = (meridiem: MeridiemType, date: Date | undefined) => {
+  return (
+    !!date && (meridiem === 'AM' ? date.getHours() < 12 : date.getHours() >= 12)
+  );
+};
+
+const formatHourFrom12 = (hour: number, meridiem: MeridiemType | undefined) => {
+  const adjustedHour = hour % 12;
+  return meridiem === 'PM' ? adjustedHour + 12 : adjustedHour;
+};
+
+const setHours = (hour: number, date: Date) => {
+  return new Date(
+    date.getFullYear(),
+    date.getMonth(),
+    date.getDate(),
+    hour,
+    date.getMinutes(),
+    date.getSeconds(),
+  );
+};
+
+export type MeridiemType = 'AM' | 'PM';
+
+export type TimePickerProps = {
+  /**
+   * Selected date. Only time is used from Date object.
+   */
+  date?: Date;
+  /**
+   * Callback when time is changed.
+   */
+  onChange?: (date: Date) => void;
+  /**
+   * Format of the time. 12h or 24h.
+   * @default false
+   */
+  use12Hours?: boolean;
+  /**
+   * Precision of the time.
+   * @default 'minutes'
+   */
+  precision?: 'hours' | 'minutes' | 'seconds';
+  /**
+   * Change step of the hours displayed.
+   * @default 1
+   */
+  hourStep?: number;
+  /**
+   * Change step of the minutes displayed.
+   * @default 1
+   */
+  minuteStep?: number;
+  /**
+   * Change step of the seconds displayed.
+   * @default 1
+   */
+  secondStep?: number;
+  /**
+   * Set focus on hour.
+   * @default false
+   */
+  setFocusHour?: boolean;
```

Comment:
How do I demo this? It seems to do nothing in storybook currently

Reply:
Ah nevermind, had to reload because I was testing smooth scrolling (which is broken). Arrow keys work